gitlab/tests/test_cli.py

```diff
@@ -29,20 +29,37 @@
 import gitlab.v4.cli
 
 
-def test_what_to_cls():
-    assert "Foo" == cli.what_to_cls("foo")
-    assert "FooBar" == cli.what_to_cls("foo-bar")
+@pytest.mark.parametrize(
+    "what,expected_class",
+    [
+        ("class", "Class"),
+        ("test-class", "TestClass"),
+        ("test-longer-class", "TestLongerClass"),
+    ],
+)
+def test_what_to_cls(what, expected_class):
+    def _namespace():
+        pass
 
+    ExpectedClass = type(expected_class, (), {})
+    _namespace.__dict__[expected_class] = ExpectedClass
 
-def test_cls_to_what():
-    class Class(object):
-        pass
+    assert cli.what_to_cls(what, _namespace) == ExpectedClass
 
-    class TestClass(object):
-        pass
 
-    assert "test-class" == cli.cls_to_what(TestClass)
-    assert "class" == cli.cls_to_what(Class)
+@pytest.mark.parametrize(
+    "class_name,expected_what",
+    [
+        ("Class", "class"),
+        ("TestClass", "test-class"),
+        ("TestUPPERCASEClass", "test-uppercase-class"),
+        ("UPPERCASETestClass", "uppercase-test-class"),
```

Comment:
Maybe add a test for

```
UserGPGKey
user-gp-gkey
```

As that was the bug.

Reply:
Added all the previously buggy scenarios, thanks!